src/main/java/bio/overture/songsearch/service/AnalysisService.java

```diff
@@ -82,4 +85,58 @@ public Analysis getAnalysisById(String analysisId) {
                             .map(Optional::get)
                             .collect(toUnmodifiableList());
  }
+
+ public List<SampleMatchedAnalysisPair> getSampleMatchedAnalysisPairs(String analysisId) {
+   val analysisFromId = getAnalysisById(analysisId);
+   val flattenedSamples = getFlattenedSamplesFromAnalysis(analysisFromId);
+   val experimentalStrategy = analysisFromId.getExperiment().get("experimental_strategy");
+
+   // short circuit return if can't find sample matched pairs for analysisFromId
+   if (experimentalStrategy == null || flattenedSamples.size() != 1) {
+     return emptyList();
+   }
+
+   val flattenedSampleOfInterest = flattenedSamples.get(0);
+   val tumourNormalDesignation = flattenedSampleOfInterest.getTumourNormalDesignation();
+
+   val filter = ImmutableMap.<String, Object>builder();
+
+   if (flattenedSampleOfInterest.getTumourNormalDesignation().equalsIgnoreCase(TUMOUR.toString())) {
+     filter.put(SUBMITTER_SAMPLE_ID, flattenedSampleOfInterest.getMatchedNormalSubmitterSampleId());
+   } else if (flattenedSampleOfInterest.getTumourNormalDesignation().equalsIgnoreCase(NORMAL.toString())) {
+     filter.put(MATCHED_NORMAL_SUBMITTER_SAMPLE_ID, flattenedSampleOfInterest.getSubmitterSampleId());
+   }
+
+   filter.put(ANALYSIS_TYPE, analysisFromId.getAnalysisType());
+   filter.put("experiment.experimental_strategy", experimentalStrategy);
+
+   return getAnalyses(filter.build(), null).stream()
+                  .map(a -> tumourNormalDesignation.equalsIgnoreCase(TUMOUR.toString()) ?
+                                    new SampleMatchedAnalysisPair(a, analysisFromId) :
+                                    new SampleMatchedAnalysisPair(analysisFromId, a))
+                  .collect(toUnmodifiableList());
+  }
```

Comment:
I moved this code out of the data fetcher and into the service level. I think it makes more sense here because there is business logic tied to this functionality.